content/en/docs/reference/labels-annotations-taints/_index.md

```diff
@@ -530,8 +530,13 @@ Example: `kubernetes.io/enforce-mountable-secrets: "true"`
 Used on: ServiceAccount
 
 The value for this annotation must be **true** to take effect.
-This annotation indicates that Pods running as this ServiceAccount may only reference
-Secret API objects specified in the ServiceAccount's `secrets` field.
+When set to "true", this annotation enforces the following rules for Pods running as this ServiceAccount:
+
+1. Secrets mounted as volumes must be listed in the ServiceAccount's `secrets` field.
+1. Secrets referenced in `envFrom` for containers or `initContainers` must also
+   be listed in the ServiceAccount's `secrets` field.
+1. Secrets referenced in a Pod's `imagePullSecrets` must be present in the
+   ServiceAccount's `imagePullSecrets` field.
 
```

Comment:
Maybe state what happens to a Pod that doesn't meet those requirements, and also when the enforcement happens.